python/nano/src/bigdl/nano/pytorch/inference/optimizer.py

```diff
@@ -218,7 +224,8 @@ def optimize(self, model: nn.Module,
                | 2. A callable object that takes model and validation_data (if
                | validation_data is not None) as input, and returns an accuracy value in
                | this calling method metric(model, data_loader) (or metric(model) if
-               | validation_data is None).
+               | validation_data is None). Note that there is no need to call `with
+               | torch.no_grad()` etc. in this object.
```

Comment:
Actually calling `torch.no_grad` won't bring any inconvenience here, it should be "there is no need to call `with InferenceOptimizer.get_context()`". Sorry for misleading.
If users call `InferenceOptimizer.get_context()`, it will be called twice (one from user, one from `optimize()`), and therefore below error will occur:
```
File "python3.7/site-packages/torch/autograd/grad_mode.py", line 297, in __exit__
    del self._inference_mode_raii_guard
AttributeError: _inference_mode_raii_guard
```
This is because `self.infer_mode.__exit__(exc_type, exc_value, exc_tb)` will execute twice.